src/daemon.rs

```diff
@@ -39,21 +55,26 @@ pub struct Daemon {
     handler_threads: HashMap<NixFile, Handler>,
     /// Sending end that we pass to every `BuildLoop` the daemon controls.
     // TODO: this needs to transmit information to identify the builder with
-    build_tx: chan::Sender<crate::build_loop::Event>,
+    build_events_tx: chan::Sender<LoopHandlerEvent>,
+    build_events_rx: chan::Receiver<LoopHandlerEvent>,
+    mon_tx: chan::Sender<LoopHandlerEvent>,
 }
 
 impl Daemon {
     /// Create a new daemon. Also return an `chan::Receiver` that
     /// receives `build_loop::Event`s for all builders this daemon
     /// supervises.
-    pub fn new() -> (Daemon, chan::Receiver<crate::build_loop::Event>) {
-        let (build_tx, build_rx) = chan::unbounded();
+    pub fn new() -> (Daemon, chan::Receiver<LoopHandlerEvent>) {
+        let (tx, rx) = chan::unbounded();
+        let (mon_tx, mon_rx) = chan::unbounded();
         (
             Daemon {
                 handler_threads: HashMap::new(),
-                build_tx,
+                build_events_tx: tx,
+                build_events_rx: rx.clone(),
```

Comment:
There should be no need to clone `rx` here. I would suggest renaming `tx` to `build_events_tx` and `rx` to `build_events_rx`, that way you can use the shorthand struct instantiation notation for `Daemon`, e.g. `Daemon { handler_threads: HashMap::new(), build_events_tx, build_events_rx, mon_tx } `.